pkg/shares/share_builder_test.go

```diff
@@ -0,0 +1,311 @@
+			want:    true,
+		},
+		{
+			name:    "continues compact share not empty",
+			builder: NewBuilder(appconsts.TxNamespaceID, appconsts.ShareVersionZero, false),
+			data:    []byte{1, 2, 3, 4, 5, 6, 7, 8, 9, 10},
+			want:    false,
+		},
+		{
+			name:    "continues sparse share not empty",
+			builder: NewBuilder([]byte{1, 1, 1, 1, 1, 1, 1, 1}, appconsts.ShareVersionZero, false),
+			data:    []byte{1, 2, 3, 4, 5, 6, 7, 8, 9, 10},
+			want:    false,
+		},
+		{
+			name:    "continues sparse share empty",
+			builder: NewBuilder([]byte{1, 1, 1, 1, 1, 1, 1, 1}, appconsts.ShareVersionZero, false),
+			data:    nil,
+			want:    true,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			tc.builder.AddData(tc.data)
+			assert.Equal(t, tc.want, tc.builder.IsEmptyShare())
+		})
+	}
+}
+
+func TestShareBuilderWriteSequenceLen(t *testing.T) {
+	type testCase struct {
+		name    string
+		builder *Builder
+		wantLen uint32
+		wantErr bool
+	}
+
+	testCases := []testCase{
+		{
+			name:    "first share",
+			builder: NewBuilder([]byte{1, 1, 1, 1, 1, 1, 1, 1}, 1, true),
+			wantLen: 10,
+			wantErr: false,
+		},
+		{
+			name:    "first share with long sequence",
+			builder: NewBuilder([]byte{1, 1, 1, 1, 1, 1, 1, 1}, 1, true),
+			wantLen: 323,
+			wantErr: false,
+		},
+		{
+			name:    "continuation sparse share",
+			builder: NewBuilder([]byte{1, 1, 1, 1, 1, 1, 1, 1}, 1, false),
+			wantLen: 10,
+			wantErr: true,
+		},
+		{
+			name:    "compact share",
+			builder: NewBuilder([]byte{0, 0, 0, 0, 0, 0, 0, 1}, 1, true),
+			wantLen: 10,
+			wantErr: false,
+		},
+		{
+			name:    "continuation compact share",
+			builder: NewBuilder([]byte{1, 1, 1, 1, 1, 1, 1, 1}, 1, false),
+			wantLen: 10,
+			wantErr: true,
+		},
+		{
+			name:    "nil builder",
+			builder: &Builder{},
+			wantLen: 10,
+			wantErr: true,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			err := tc.builder.WriteSequenceLen(tc.wantLen)
+			if tc.wantErr {
+				assert.Error(t, err)
+				return
+			}
+
+			tc.builder.ZeroPadIfNecessary()
+			share, err := tc.builder.Build()
+			require.NoError(t, err)
+
+			len, err := share.SequenceLen()
+			require.NoError(t, err)
+
+			assert.Equal(t, tc.wantLen, len)
+		})
+	}
+}
+
+func TestShareBuilderAddData(t *testing.T) {
+	type testCase struct {
+		name    string
+		builder *Builder
+		data    []byte // input data
+		want    []byte
+	}
+
+	testCases := []testCase{
+		{
+			name:    "small share",
+			builder: NewBuilder([]byte{1, 1, 1, 1, 1, 1, 1, 1}, appconsts.ShareVersionZero, true),
+			data:    []byte{1, 2, 3, 4, 5, 6, 7, 8, 9, 10},
+			want:    nil,
+		},
+		{
+			name:    "exact fit first compact share",
+			builder: NewBuilder(appconsts.TxNamespaceID, appconsts.ShareVersionZero, true),
+			data:    bytes.Repeat([]byte{1}, appconsts.ShareSize-appconsts.NamespaceSize-appconsts.CompactShareReservedBytes-appconsts.SequenceLenBytes-1 /*1 = info byte*/),
```

Comment:
[nit] I think the tests in this file can use the const `appconsts.ShareInfoBytes` instead of `-1 /*1 = info byte*/`.

```suggestion
			data:    bytes.Repeat([]byte{1}, appconsts.ShareSize-appconsts.NamespaceSize-appconsts.ShareInfoBytes-appconsts.CompactShareReservedBytes-appconsts.SequenceLenBytes),
```